From a99a2c3d6e48ac35ce234045cd70b9b1a7e3dec4 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Sat, 7 Oct 2017 03:28:44 -0400
Subject: [PATCH 3/3] python: fix library load and install

This prevents installing an additional copy of the mxnet shared library
and fixes the loading of the library from PATH.

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 python/mxnet/base.py | 3 +--
 python/setup.py      | 2 --
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/python/mxnet/base.py b/python/mxnet/base.py
index fe80bd69..1c512f15 100644
--- a/python/mxnet/base.py
+++ b/python/mxnet/base.py
@@ -98,8 +98,7 @@ class MXCallbackList(ctypes.Structure):
 
 def _load_lib():
     """Load library by searching possible path."""
-    lib_path = libinfo.find_lib_path()
-    lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
+    lib = ctypes.CDLL("libmxnet.so", ctypes.RTLD_LOCAL)
     # DMatrix functions
     lib.MXGetLastError.restype = ctypes.c_char_p
     return lib
diff --git a/python/setup.py b/python/setup.py
index 14c8121d..4c435f93 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -43,7 +43,6 @@ libinfo_py = os.path.join(CURRENT_DIR, 'mxnet/libinfo.py')
 libinfo = {'__file__': libinfo_py}
 exec(compile(open(libinfo_py, "rb").read(), libinfo_py, 'exec'), libinfo, libinfo)
 
-LIB_PATH = libinfo['find_lib_path']()
 __version__ = libinfo['__version__']
 
 
@@ -103,7 +102,6 @@ setup(name='mxnet',
       version=__version__,
       description=open(os.path.join(CURRENT_DIR, 'README.md')).read(),
       packages=find_packages(),
-      data_files=[('mxnet', [LIB_PATH[0]])],
       url='https://github.com/apache/incubator-mxnet',
       ext_modules=config_cython(),
       **kwargs)
diff --git a/mxnet-9999/mshadow/cmake/mshadow.cmake b/mxnet-9999/mshadow/cmake/mshadow.cmake_
index 0aac6f8..ffd4a77 100644
--- a/mshadow/cmake/mshadow.cmake
+++ b/mshadow/cmake/mshadow.cmake
@@ -1,6 +1,6 @@
 set(mshadow_LINKER_LIBS "")
 
-set(BLAS "Open" CACHE STRING "Selected BLAS library")
+set(BLAS "Atlas" CACHE STRING "Selected BLAS library")
 set_property(CACHE BLAS PROPERTY STRINGS "Atlas;Open;MKL")
 
 if(USE_MKL_IF_AVAILABLE)
--- a/cmake/ChooseBlas.cmake
+++ b/cmake/ChooseBlas.cmake
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-set(BLAS "Open" CACHE STRING "Selected BLAS library")
+set(BLAS "Atlas" CACHE STRING "Selected BLAS library")
 set_property(CACHE BLAS PROPERTY STRINGS "Atlas;Open;MKL")
 
 if(USE_MKL_IF_AVAILABLE)
